iri-commons/src/main/java/no/sikt/nva/iri/SerialPublicationId.java

```diff
@@ -0,0 +1,31 @@
+package no.sikt.nva.iri;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import java.net.URI;
+import java.time.Year;
+import java.util.UUID;
+
+public record SerialPublicationId(UUID identifier, Year year) implements PublicationChannelId {
+
+    private static final ChannelType CHANNEL_TYPE = ChannelType.SERIAL_PUBLICATION;
+
+    public SerialPublicationId(UuidYearPair validate) {
+        this(validate.uuid(), validate.year());
+    }
+
+    public static SerialPublicationId from(URI uri) {
+        return new SerialPublicationId(PublicationChannelId.validate(uri, CHANNEL_TYPE));
```

Comment:
Validate method implies void or boolean, but here we return an object.
What about do validation and then extract identifier and year by methods in common interface?


```
  public static SerialPublicationId from(URI uri) {
      PublicationChannelId.validate(uri, CHANNEL_TYPE);
      var identifier = PublicationChannelId.extractIdentifier(uri);
      var year = PublicationChannelId.extractYear(uri);
      return new SerialPublicationId(identifier, year);
  }
```
Thank we may no need wrapper object `UuidYearPair`